### PR TITLE
[SPARK-39492][SQL] Rework MISSING_COLUMN

### PR DESCRIPTION
**Pull request by srielau**
### What changes were proposed in this pull request?
Replace the error class MISSING_COLUMN with UNRESOLVED_COLUMN wit the following new text:

A column or function parameter with name <objectName> cannot be resolved. Did you mean one of the following? [<objectList>]

### Why are the changes needed?

The existing class name and text does not reflect what is really happening. The column may well exist, it may just not be within scope.

### Does this PR introduce _any_ user-facing change?

Yes, we we replace an error class name.

### How was this patch tested?

test all affected test suites with MISSING_COLUMN

### CODE REVIEW

File: python/pyspark/sql/tests/test_utils.py [hunk 1]
```diff
@@ -36,7 +36,9 @@ def test_capture_user_friendly_exception(self):
         try:
             self.spark.sql("select `中文字段`")
         except AnalysisException as e:
-            self.assertRegex(str(e), "Column '`中文字段`' does not exist")
+            self.assertRegex(str(e),
+                             "[UNRESOLVED_COLUMN] A column or function parameter with name " +
+                             "`中文字段` cannot be resolved. Did you mean one of the following? []")
```
Comment (srielau) [inline python/pyspark/sql/tests/test_utils.py:41]:
```suggestion
            self.assertRegex(
                str(e),
                "[UNRESOLVED_COLUMN] A column or function parameter with name " +
                "`中文字段` cannot be resolved. Did you mean one of the following? []"
            )
```

**Review by gengliangwang — APPROVED** [review]
The error message looks clear now!

**Comment by gengliangwang** [comment]
https://github.com/srielau/spark/runs/6942384510?check_suite_focus=true

@srielau I guess you will have to retrigger the tests

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**